### PR TITLE
[terra-clinical-item-view] Implement item view as unordered list

### PR DESCRIPTION
**Pull request by RayGunY**
Note: This PR is a little messy right now as it's pointing towards UXPLATFORM-9563. Once that's merged this will be able to be rebased with main and it'll clean up better.

### Summary

**What was changed:**
- Implementing the usage of unordered lists to contain the displays as list items
- Split out the logic into more individual methods for readability/because there were so many different scenarios to handle
- add guidance for consumers about adding a header for the item view

For two column layout (this includes the byRow formatting) the implementation of the unordered lists is slightly different. It works as an unordered list that contains the two columns as list items. Each column is an unordered list as well that contains the displays as list items.

_For normal two column layout:_
 Unordered  List
-> list item 1 (column 1 as UL) -> column 1's displays as list items
-> list item 2 (column 2 as UL) -> column 2's displays as list items

_For by row two column layout:_
Unordered List
-> list item 1 (row 1 as UL) -> row 1's two displays as list items
-> list item 2 (row 2 as UL) -> row 2's two displays as list items
-> list item 3 (row 3 as UL) -> row 3's two displays as list items
-> list item 4 (row 4 as UL) -> row 4's two displays as list items

When there is just a single display passed in to the item view we don't treat it as part of a list.

**Why it was changed:**
This was changes so we have greater context when using the screen reader of how this information is grouped.

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details
Validation posted below, first two are how the implementation will read when the consumers adds their header. Also added a recording of the 'by row' readout since it is different than the other two.

https://github.com/cerner/terra-clinical/assets/40574651/01f0aed2-295e-4132-a864-008b6909b23b

https://github.com/cerner/terra-clinical/assets/40574651/9277ff13-18b7-4388-a5e9-aed899b46f73

https://github.com/cerner/terra-clinical/assets/40574651/3e4aa3c1-5603-40f4-9a2c-3f46f216672c

https://github.com/cerner/terra-clinical/assets/40574651/5435fb24-660f-4862-a81b-226304a594b2

https://github.com/cerner/terra-clinical/assets/40574651/e669bcfc-7506-458c-90e6-37655f295530

https://github.com/cerner/terra-clinical/assets/40574651/8b8bd224-fd58-426f-b4e7-e37e1db26a1c

**This PR resolves:**

UXPLATFORM-9430 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-clinical-item-view/src/ItemView.jsx [hunk 1]
```diff
@@ -145,6 +146,114 @@ const classesForContent = (rowIndex, rowCount, contentIndex, emphasis) => {
...
+    </div>
+  );
+};
+
+const renderSingleDisplayView = (singleDisplay, overrideDefaultStyling) => {
+  /**
+   * Since this is always a singular display, the content styling will be the primary defaults if they are not overridden.
+   * We don't have to call into the classesForContent method and instead can just set the primary size and color here.
+   */
+  const contentClass = overrideDefaultStyling ? 'content' : ['content'].concat(['content-primary-size', 'content-primary-color']);
```
Comment (zxeleven) [inline packages/terra-clinical-item-view/src/ItemView.jsx:246]: nit: do we need to use `concat` here or can we just use the following?
```
['content', 'content-primary-size', 'content-primary-color']
```

File: packages/terra-clinical-item-view/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,9 @@
 * Added
   * Added `trueColumn` prop so consumers can choose whether to separate displays in the two column layout by row or by column.
 
+* Changed
+  * Changed displays to exist in an unordered list as list items with the first display as a programatic header.
```
[the remark this answers is not in the document]
Comment (RayGunY) [reply packages/terra-clinical-item-view/CHANGELOG.md]: [fix spelling](https://github.com/cerner/terra-clinical/pull/913/commits/cc2a7ba341486ae226c1e93531965820ec7a0304)

File: packages/terra-clinical-item-view/src/ItemView.jsx (hunk 1 above)
Comment (RayGunY) [reply packages/terra-clinical-item-view/src/ItemView.jsx]: Good suggestion, changed here: [073a1ab](https://github.com/cerner/terra-clinical/pull/913/commits/073a1abb106f1bceb6672e2cd92ad8ce27fa17b2)

File: packages/terra-clinical-item-view/tests/jest/ItemView.test.jsx [hunk 3]
```diff
@@ -3,6 +3,13 @@ import React from 'react';
 import { shallowWithIntl } from 'terra-enzyme-intl';
 import ItemView from '../../lib/ItemView';
 
+jest.mock('uuid', () => ({ v4: () => '00000000-0000-0000-0000-000000000000' }));
```
Comment (vinaybhargavar) [inline packages/terra-clinical-item-view/tests/jest/ItemView.test.jsx:6]: I believe recently there was a [PR](https://github.com/cerner/terra-clinical/pull/911/files#diff-5c8f054737c8e43c7ebd6f698efb698903d7bf556d1add6c6ab24425acbb5f47R64-R71) to change jest.mock to jest.spyOn. Can we use the spyOn here as well?

File: packages/terra-clinical-item-view/tests/jest/ItemView.test.jsx [hunk 4]
```diff
@@ -3,6 +3,13 @@ import React from 'react';
...
+
+// Note: when multiple displays are passed in to the ItemView, the length of 'ItemDisplay' in these tests will
```
Comment (zxeleven) [inline packages/terra-clinical-item-view/tests/jest/ItemView.test.jsx:8]: nit: can we use a block comment here?

**Review by zxeleven — APPROVED** [review]
Looks good, just one small nit

**Comment by RayGunY** [comment]
@eawww @sycombs @sdadn this is ready to be looked when you've got a moment

**Review by vinaybhargavar — APPROVED** [review]

**Comment by RayGunY** [comment]
Posting in it's own comment - I was having trouble getting the header label to read out with JAWs. Dug around a bit and JAWs by default does not read out headers for lists. More on that here: https://stackoverflow.com/questions/36882441/jaws-not-reading-listview-column-headers

So since by default it doesn't read them out (presumably for some reason that the JAWs team has determined) this should be good in that regard. Just to be as thorough as possible I've been trying to figure out how to switch the settings so it will read out the header, but I've had no luck so far. I've tried to do these steps but I can't access the manager it seems? https://docs.oracle.com/applications/smartview/610/SVAAA/enabling_jaws_to_read_listview_headers.htm#SVAAA-smart_view_accessibility_book_4

I've posted recordings to at least show that it's reading them out as a list now. Does the above JAWs behavior sound reasonable? Not sure if I'll be able to get recordings with that setting working.

**Comment by sdadn** [comment]
Does this depend on #912 ? If so, I'll review it once that gets merged to avoid messy diffs and merges.

**Comment by sdadn** [comment]
#912 Has been merged, once you update it with main, I can review this.

**Review by scottwilmarth — APPROVED** [review]
There are still improvements that should be made, but there is not a clear path to manage the needed changes when considering passivity. These updates are not perfect, but they represent a better path and will be great to ensure better support for accessibility.

**Review by sycombs — APPROVED** [review]

**Review by sdadn — APPROVED** [review]

**Pull request merged**